### PR TITLE
FIX: Ensure we clean up socket servers.

### PR DESCRIPTION
**Pull request by tgxworld**
cc @SamSaffron  Can you review this? Thank you!

### CODE REVIEW

**Comment by SamSaffron** [comment]
hmmm... we need a test here if we are changing internals. 

I guess this ensures that if the container restarts and pids end up being reused we unconditionally clean up, but I though unicorn already cleaned up on boot? 

need more context about the actual bug being fixed here, also I feel this change is at high risk of leaving orphans around cause there is no cleanup anymore.

**Comment by tgxworld** [comment]
> I guess this ensures that if the container restarts and pids end up being reused we unconditionally clean up, but I though unicorn already cleaned up on boot?

@SamSaffron  This is related to the question I asked you about on Slack yesterday. Unicorn only cleans up the files for the unicorn workers. The files for unicorn master and Sidekiq are not being nuked when the thread dies so I made this change to give us a clean pattern of cleaning up. My believe is that the `SocketServer` should be responsible of cleaning up itself once it stops. Also, we're not calling `@server.close` at all for any of the `SocketServer` that we start so I was wondering if that might leave us in a weird state.

> also I feel this change is at high risk of leaving orphans around cause there is no cleanup anymore.

Hmm which orphans are you referring to? All files in the stats directory are nuked on boot should the cleanup fail.

File: lib/socket_server.rb [hunk 1]
```diff
@@ -19,16 +19,22 @@ def stop
     @server.close if @server
     @server = nil
     @blk = nil
+    FileUtils.rm_f(@socket_path)
```
Comment (SamSaffron) [inline lib/socket_server.rb]: need to protect against errors here, cause we have @server.close if @server ... similarly only rm if the socket path is there

**Comment by SamSaffron** [comment]
Sure seems safe enough!

**Pull request merged**